lib/modules/showImages.js

```diff
@@ -810,6 +810,15 @@ function resolveMediaUrl(element, thing) {
 		thing &&
 		element.classList.contains('title')
 	) {
+		// In old.reddit.com, a reddit gallery's title link can be misleadingly set
+		// to the source link in the gallery's caption instead of the normal reddit.com/gallery/<id> URL.
+		// The only reliable gallery detection is the "data-is-gallery" attribute in the Thing element.
+		const isGallery = thing.element.getAttribute('data-is-gallery') === 'true';
+		if (isGallery) {
```

Comment:
Could probably be simplified to `if (thing.element.dataset.isGallery)`, otherwise looking good!

Reply:
I inlined the boolean expression, but I can't do a truthy string check on `dataset.isGallery` alone because `dataset.isGallery` can be set to a `"false"` string.